registry/remote/example_test.go

```diff
@@ -0,0 +1,326 @@
+const exampleTag = "latest"
+const exampleBlob = "Example blob content"
+const exampleUploadUUid = "0bc84d80-837c-41d9-824e-1907463c53b3"
+
+var host string
+
+func TestMain(m *testing.M) {
+	// Setup a local HTTPS registry
+	ts := httptest.NewTLSServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		p := r.URL.Path
+		m := r.Method
+		switch {
+		case p == "/v2/_catalog" && m == "GET":
+			result := struct {
+				Repositories []string `json:"repositories"`
+			}{
+				Repositories: []string{"public/repo1", "public/repo2", "internal/repo3"},
+			}
+			json.NewEncoder(w).Encode(result)
+		case p == fmt.Sprintf("/v2/%s/tags/list", exampleRepositoryName) && m == "GET":
+			result := struct {
+				Tags []string `json:"tags"`
+			}{
+				Tags: []string{"tag1", "tag2"},
+			}
+			json.NewEncoder(w).Encode(result)
+		case p == fmt.Sprintf("/v2/%s/blobs/uploads/", exampleRepositoryName):
+			w.Header().Set("Location", p+exampleUploadUUid)
+			w.Header().Set("Docker-Upload-UUID", exampleUploadUUid)
+			w.WriteHeader(http.StatusAccepted)
+		case p == fmt.Sprintf("/v2/%s/blobs/uploads/%s", exampleRepositoryName, exampleUploadUUid):
+			w.WriteHeader(http.StatusCreated)
+		case (p == fmt.Sprintf("/v2/%s/manifests/latest", exampleRepositoryName) || p == fmt.Sprintf("/v2/%s/manifests/%s", exampleRepositoryName, exampleDigest)) && m == "HEAD":
+			w.Header().Set("Content-Type", ocispec.MediaTypeImageLayer)
+			w.Header().Set("Docker-Content-Digest", exampleDigest)
+			w.Header().Set("Content-Length", strconv.Itoa(len([]byte(exampleBlob))))
+		case (p == fmt.Sprintf("/v2/%s/manifests/%s", exampleRepositoryName, exampleDigest)) && m == "GET":
+			w.Header().Set("Content-Type", ocispec.MediaTypeImageLayer)
+			w.Header().Set("Docker-Content-Digest", exampleDigest)
+			w.Header().Set("Content-Length", strconv.Itoa(len([]byte(exampleBlob))))
+			w.Write([]byte(exampleBlob))
+		case p == fmt.Sprintf("/v2/%s/blobs/%s", exampleRepositoryName, exampleDigest) && m == "GET":
+			w.Header().Set("Content-Type", ocispec.MediaTypeImageLayer)
+			w.Header().Set("Docker-Content-Digest", exampleDigest)
+			w.Header().Set("Content-Length", strconv.Itoa(len([]byte(exampleBlob))))
+			w.Write([]byte(exampleBlob))
+		case p == fmt.Sprintf("/v2/%s/manifests/%s", exampleRepositoryName, exampleTag) && m == "PUT":
+			w.WriteHeader(http.StatusCreated)
+		}
+
+	}))
+	defer ts.Close()
+	u, err := url.Parse(ts.URL)
+	if err != nil {
+		panic(err) // Handle error
+	}
+	host = u.Host
+	http.DefaultClient = ts.Client()
+
+	os.Exit(m.Run())
+}
+
+// ExampleRepository_Tags gives example snippets for listing tags in a repository.
+func ExampleRepository_Tags() {
+	reg, err := remote.NewRegistry(host)
+	if err != nil {
+		panic(err) // Handle error
+	}
+
+	ctx := context.Background()
+	repo, err := reg.Repository(ctx, exampleRepositoryName) // Get the repository from registry
+	if err != nil {
+		panic(err) // Handle error
+	}
+
+	repo.Tags(ctx, func(tags []string) error {
```

Comment:
error check required.

Reply:
Good catch. Will the function return non-nil error even when the callback only returns nil?